lib/cgo/cipher.crypto.go

```diff
@@ -0,0 +1,278 @@
+	slice := inplacePubKeySlice(_slice)
+	return slice.Len()
+}
+
+//export SKY_cipher_PubKeySlice_Less
+func SKY_cipher_PubKeySlice_Less(_slice *C.PubKeySlice, _i, _j int) bool {
+	slice := inplacePubKeySlice(_slice)
+	return slice.Less(_i, _j)
+}
+
+//export SKY_cipher_PubKeySlice_Swap
+func SKY_cipher_PubKeySlice_Swap(_slice *C.PubKeySlice, _i, _j int) {
+	slice := inplacePubKeySlice(_slice)
+	slice.Swap(_i, _j)
+}
+
+//export SKY_cipher_RandByte
+func SKY_cipher_RandByte(_n int, _arg1 *C.GoSlice_) {
+	b := cipher.RandByte(_n)
+	copyToGoSlice(reflect.ValueOf(b), _arg1)
+}
+
+//export SKY_cipher_NewPubKey
+func SKY_cipher_NewPubKey(_b []byte, _arg1 *C.PubKey) (errcode uint32) {
+	defer func() {
+		errcode = catchApiPanic(errcode, recover())
+	}()
+
+	pubkey := cipher.NewPubKey(_b)
+	copyToBuffer(reflect.ValueOf(pubkey[:]), unsafe.Pointer(_arg1), uint(SizeofPubKey))
+	return libErrorCode(nil)
+}
+
+//export SKY_cipher_PubKeyFromHex
+func SKY_cipher_PubKeyFromHex(_s string, _arg1 *C.PubKey) (errcode uint32) {
+	defer func() {
+		errcode = catchApiPanic(errcode, recover())
+	}()
+
+	pubkey, err := cipher.PubKeyFromHex(_s)
+	errcode = libErrorCode(err)
+	if err == nil {
+		copyToBuffer(reflect.ValueOf(pubkey[:]), unsafe.Pointer(_arg1), uint(SizeofPubKey))
+	}
+	return errcode
+}
+
+//export SKY_cipher_PubKeyFromSecKey
+func SKY_cipher_PubKeyFromSecKey(_seckey *C.SecKey, _arg1 *C.PubKey) (errcode uint32) {
+	defer func() {
+		errcode = catchApiPanic(errcode, recover())
+	}()
+
+	seckey := (*cipher.SecKey)(unsafe.Pointer(_seckey))
+
+	pubkey := cipher.PubKeyFromSecKey(*seckey)
+
+	copyToBuffer(reflect.ValueOf(pubkey[:]), unsafe.Pointer(_arg1), uint(SizeofPubKey))
+	return libErrorCode(nil)
+}
+
+//export SKY_cipher_PubKeyFromSig
+func SKY_cipher_PubKeyFromSig(_sig *C.Sig, _hash *C.SHA256, _arg2 *C.PubKey) uint32 {
+	sig := (*cipher.Sig)(unsafe.Pointer(_sig))
+	hash := (*cipher.SHA256)(unsafe.Pointer(_hash))
+
+	pubkey, err := cipher.PubKeyFromSig(*sig, *hash)
+
+	errcode := libErrorCode(err)
+	if err == nil {
+		copyToBuffer(reflect.ValueOf(pubkey[:]), unsafe.Pointer(_arg2), uint(SizeofPubKey))
+	}
+	return errcode
+}
+
+//export SKY_cipher_PubKey_Verify
+func SKY_cipher_PubKey_Verify(_pk *C.PubKey) uint32 {
+	pk := (*cipher.PubKey)(unsafe.Pointer(_pk))
+
+	err := pk.Verify()
+	errcode := libErrorCode(err)
+	return errcode
+}
+
+//export SKY_cipher_PubKey_Hex
+func SKY_cipher_PubKey_Hex(_pk *C.PubKey, _arg1 *C.GoString_) {
+	pk := (*cipher.PubKey)(unsafe.Pointer(_pk))
+	s := pk.Hex()
+	copyString(s, _arg1)
+}
+
+//export SKY_cipher_PubKey_ToAddressHash
+func SKY_cipher_PubKey_ToAddressHash(_pk *C.PubKey, _arg0 *C.Ripemd160) {
+	pk := (*cipher.PubKey)(unsafe.Pointer(_pk))
+	h := pk.ToAddressHash()
+	copyToBuffer(reflect.ValueOf(h[:]), unsafe.Pointer(_arg0), uint(SizeofRipemd160))
+}
+
+//export SKY_cipher_NewSecKey
+func SKY_cipher_NewSecKey(_b []byte, _arg1 *C.SecKey) (errcode uint32) {
+	defer func() {
+		errcode = catchApiPanic(errcode, recover())
```

Comment:
Is there a problem with allowing it to panic?

The reason it panics is because if the condition ever occurs, the code should not be used at all or else there is a risk of fatal cryptographic errors